### PR TITLE
Added API

### PR DESCRIPTION
**Pull request by begedin**
- basic ping endpoint, with unauthenticated and authenticated response
- doorkeeper + clearance for authentication
- basic get /users/:id endpoint, AMS enabled

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Documentation:
+  Enabled: false
+Style/StringLiterals:
+  EnforcedStyle: double_quotes
```
Comment (christopherstyles) [inline .rubocop.yml:4]: Any chance we can stick with single quotes? Only speaking for myself here, the reasons why this is important: 
- The difference adds a semantic benefit when scanning code, to quickly recognize what strings are interpolated. 
- The choice of single vs. double conveys intention.
- Editors can pick up on the difference and syntax highlight accordingly. 
- Provides an additional matching pattern “hook”, upon which to grep. 

This is obviously my own preference, and I definitely understand not all folks benefit from such things. Thoughts?

Comment (joshsmith) [inline .rubocop.yml:4]: We went with Thoughtbot's reasoning on this, which was that there are no performance benefits to double vs single and therefore you don't have to think about escaping or interpolating.

Comment (joshsmith) [inline .rubocop.yml:4]: It was a little confusing the first time we used Hound CI and were used to your style. Ultimately bought into their thinking behind it and haven't looked back.

Comment (christopherstyles) [inline .rubocop.yml:4]: Performance is surely the last reason I would consider, given the choice of single vs. double quotes. For me it really is primarily about conveying _intention_, and having a couple additional tools at your disposal for quickly scanning code, or grepping.

Comment (christopherstyles) [inline .rubocop.yml:4]: I agree with about 99.5% of what Thoughtbot says. :stuck_out_tongue: Sort of surprised at this one, considering they’re a vim shop. For me, the extra keystroke (Shift) for a double quote suggests an intentionally “special” string, matching up with the thought of declaring it interpolated. This is a reasonable/negligible overhead imo, for the added benefits.

But anyhow, I digress – if everyone else on the team prefers double quotes then we should definitely go with that.

File: db/migrate/20160324133505_add_clearance_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class AddClearanceToUsers < ActiveRecord::Migration
+  def self.up
+    change_table :users do |t|
+      t.string :encrypted_password, limit: 128
+      t.string :confirmation_token, limit: 128
+      t.string :remember_token, limit: 128
+    end
+
+    add_index :users, :email
```
Comment (christopherstyles) [inline db/migrate/20160324133505_add_clearance_to_users.rb]: Should this also be unique?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe "schema" do
+    it { should have_db_column(:email).of_type(:string).with_options(null: false) }
+    it { should have_db_column(:encrypted_password).of_type(:string) }
+
+    it { should have_db_column(:updated_at) }
+    it { should have_db_column(:created_at) }
```
Comment (christopherstyles) [inline spec/models/user_spec.rb:9]: Can we stick to `expect` syntax? We also probably want to set this in the spec_helper configuration:

``` ruby
RSpec.configure do |config|
  config.expect_with :rspec do |expectations|
    # Enable only the newer, non-monkey-patching expect syntax.
    # For more details, see:
    #   - http://myronmars.to/n/dev-blog/2012/06/rspecs-new-expectation-syntax
    expectations.syntax = :expect
  end
end
```

**Comment by christopherstyles** [comment]
This is looking great @begedin, nice work! Have we given any thought to API versioning yet?

File: spec/models/user_spec.rb (hunk 3 above)
Comment (joshsmith) [inline spec/models/user_spec.rb:9]: Generally, yes. According to shoulda matchers, though, this should still work regardless of how rspec is configured. We could do `is_expected.to` as an alternative.

File: db/migrate/20160324133505_add_clearance_to_users.rb (hunk 2 above)
Comment (joshsmith) [inline db/migrate/20160324133505_add_clearance_to_users.rb]: Yes, it should.

**Comment by joshsmith** [comment]
@christopherstyles we actually started talking about API versioning the other day. With Field the Bern, we made no assumptions and just decided to tread kind of recklessly with regards to versioning.

I'd been reading a lot about how Stripe does their versioning using gates and request/response layers with date headers instead of hardcoded URL versions.

We may not (at first) have many consumers beyond our own few clients. So we should balance decisions here with an eye towards understanding resource constraints and longevity. We may have more than three clients at some point. And we may have more than us consuming the API. We also don't have a ton of time to devote to a complex scheme, though understand similarly that we want to be able to iterate in fast feedback cycles, so a date-based approach can make more sense.

Do you have any thoughts here? We should maybe open a separate issue with my comments pasted in.

**Comment by christopherstyles** [comment]
:+1:

**Comment by joshsmith** [comment]
@begedin missing a reviewed label but :+1:

**Pull request merged**